### PR TITLE
sergi-e/linux-build

### PR DESCRIPTION
**Pull request by sergi-e**
#### Description
- Linux build installation process has been rewritten. 

#### Where has this been tested?

  * **Platform(s):** Linux 18.04.3
  * **Python version(s):** Python 2.7.17 and 3.6.9
  * **Unreal Engine version(s):** 4.22

#### Possible Drawbacks
 - clang paragraph could be improved. 
- A specific command to set environment variable persistently could be added. 
- Test if it is better to add "&&" before the break lines for separated command sets.

---
This change is [Reviewable](https://reviewable.io/reviews/carla-simulator/carla/2412)

### CODE REVIEW

File: Docs/how_to_build_on_linux.md [hunk 1]
```diff
@@ -1,136 +1,194 @@
-<h1>How to build CARLA on Linux</h1>
----
+<h1>Linux build</h1>
 
-!!! note
-    CARLA requires Ubuntu 16.04 or later.
 
-Install the build tools and dependencies
+* [Requirements](#requirements):
```
Comment (glopezdiest) [inline Docs/how_to_build_on_linux.md]: "Minimum to run CARLA" and "Needed to build the project" gives no information. Either say it specifically or erase the subparts.

In general, I'd leave the subparts of the index, but erase their explanation, as most of them are redundant.
If needed, remark them in bold in their sections?

File: Docs/how_to_build_on_linux.md [hunk 2]
```diff
@@ -1,136 +1,194 @@
...
+<a href="https://forum.carla.org/" target="_blank" class="btn btn-neutral" title="Go to the latest CARLA release">
+CARLA forum</a>
+</p>
+</div>
+
+---------------
+##Requirements
+
+<h4>System specifics</h4>
+* __Ubuntu 16.04 or later:__ migration to Ubuntu 18 is slowly coming forward, but today's date there is still support for 16.04.
```
Comment (glopezdiest) [inline Docs/how_to_build_on_linux.md]: Weird sentence. What about:

Currently migrating to Ubuntu 18.04, but version 16.04 is still being supported.

File: Docs/how_to_build_on_linux.md [hunk 3]
```diff
@@ -1,136 +1,194 @@
...
+
+<h4>System specifics</h4>
+* __Ubuntu 16.04 or later:__ migration to Ubuntu 18 is slowly coming forward, but today's date there is still support for 16.04. 
+* __Two TCP ports:__ 2000 and 1 by default. Be sure neither firewall nor any other application are blocking these. 
+* __An adequate GPU:__ CARLA aims for realistic simulations, so the server needs at least a 4GB GPU. A dedicated GPU is highly recommended for machine learning. 
+
+
+<h4>Dependencies</h4>
+
+CARLA needs many dependencies to run. Some of them are built automatically during this process, such as _Boost.Python_. Others are binaries that should to be installed before starting the build (_cmake_, _clang_, different versions of _Python_ and much more). In order to do so, run the commands below in a terminal window.
```
Comment (glopezdiest) [inline Docs/how_to_build_on_linux.md]: remove the "to" in should to be installed

File: Docs/how_to_build_on_linux.md [hunk 4]
```diff
@@ -1,136 +1,194 @@
...
-sudo apt-get update
-sudo apt-get install build-essential clang-7 lld-7 g++-7 cmake ninja-build libvulkan1 python python-pip python-dev python3-dev python3-pip libpng16-dev libtiff5-dev libjpeg-dev tzdata sed curl unzip autoconf libtool rsync
-pip2 install --user setuptools
-pip3 install --user setuptools
 ```
 
-!!! tip
-    For **Ubuntu 18.04**, change `libpng16-dev` to `libpng-dev` from the previous example.
+!!! important
+    The following commands differ depending on Ubuntu version. The only change is `libpng16-dev` which becomes `libpng-dev`, but the full set of commands is here twice to ease copying them.
```
Comment (glopezdiest) [inline Docs/how_to_build_on_linux.md]: While the only change is `libpng16-dev` (which becomes `libpng-dev`), the full set of commands is here for easier copying.

File: Docs/how_to_build_on_linux.md [hunk 5]
```diff
@@ -1,136 +1,194 @@
...
+
+The last step is to finally build CARLA. There are different `make` commands to build the different modules. All of them run in the root CARLA folder: 
+
+* **launch** will compile the server simulator and launch it in Unreal Engine. Press **Play** to start the spectator view. Camera can be moved with WASD keys and rotated by clicking the scene while moving the mouse around. Close the editor window to exit:  
+```sh
+make launch
+``` 
+* **PythonAPI** will compile the API client, necessary to grant control over the simulation. After building it, scripts can run. The following example will run a dynamic weather on the scene: 
+```sh
+make PythonAPI && cd PythonAPI/examples && ./dynamic_weather.py
```
Comment (glopezdiest) [inline Docs/how_to_build_on_linux.md]: Maybe a more visual example related to the cars? like spawn_npc.py?

Comment (sergi-e) [reply Docs/how_to_build_on_linux.md]: @glopezdiest Updated the file with corresponding suggestions in mind.

**Comment by a-jahani** [comment]
What's the problem with the current Linux build instructions?

**Pull request merged**